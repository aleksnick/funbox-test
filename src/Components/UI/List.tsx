import React from "react";
import UIList from "@material-ui/core/List";

export interface ListProps {}

/**
 * Кнопка с иконкой
 *
 * @export
 * @class List
 * @extends {React.Component<ListProps>}
 */
export default class List extends React.Component<ListProps> {
  constructor(props: ListProps) {
    super(props);
  }

  render() {
    return <UIList>{this.props.children}</UIList>;
  }
}
