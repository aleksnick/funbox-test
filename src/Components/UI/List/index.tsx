import React from "react";
import UIList from "@material-ui/core/List";
import ICanStyled from "../../../Models/ICanStyled";

export interface ListProps extends ICanStyled {}

/**
 * Список
 *
 * @export
 * @class List
 * @extends {React.Component<ListProps>}
 */
export default class List extends React.Component<ListProps> {
  constructor(props: ListProps) {
    super(props);
  }

  render() {
    const { className, children } = this.props;
    return <UIList className={className}>{children}</UIList>;
  }
}
