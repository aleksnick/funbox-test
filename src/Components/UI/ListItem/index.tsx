import React from "react";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import Paper from "@material-ui/core/Paper";
import ICanStyled from "../../../Models/ICanStyled";

export interface ListProps extends ICanStyled {
  extraContent?: React.ReactNode;
}

/**
 * Элемент списка
 *
 * @export
 * @class List
 * @extends {React.Component<ListProps>}
 */
export default class List extends React.Component<ListProps> {
  constructor(props: ListProps) {
    super(props);
  }

  render() {
    const { className, children, extraContent } = this.props;
    return (
      <Paper className={className}>
        <ListItem>
          <ListItemText primary={children} />
          <ListItemSecondaryAction>{extraContent}</ListItemSecondaryAction>
        </ListItem>
      </Paper>
    );
  }
}
