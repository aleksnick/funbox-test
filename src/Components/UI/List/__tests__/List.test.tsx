import React from "react";
import List from "../";
import ListItem from "../../ListItem";
import { render } from "enzyme";

describe("List", () => {
  describe("Snapshots", () => {
    test("Simple List snapshot", () => {
      const wrapper = render(
        <List>
          <ListItem key={0}>One</ListItem>
          <ListItem key={1}>Two</ListItem>
          <ListItem key={2}>Three</ListItem>
        </List>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
