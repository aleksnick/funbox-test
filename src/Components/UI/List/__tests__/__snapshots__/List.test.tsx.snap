// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snapshots Simple List snapshot 1`] = `
<ul
  class="MuiList-root-1 MuiList-padding-2"
>
  <div
    class="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
  >
    <li
      class="MuiListItem-container-33"
    >
      <div
        class="MuiListItem-root-32 MuiListItem-default-35 MuiListItem-gutters-40 MuiListItem-secondaryAction-42"
      >
        <div
          class="MuiListItemText-root-44"
        >
          <span
            class="MuiTypography-root-50 MuiTypography-subheading-57 MuiListItemText-primary-47"
          >
            One
          </span>
        </div>
      </div>
      <div
        class="MuiListItemSecondaryAction-root-86"
      />
    </li>
  </div>
  <div
    class="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
  >
    <li
      class="MuiListItem-container-33"
    >
      <div
        class="MuiListItem-root-32 MuiListItem-default-35 MuiListItem-gutters-40 MuiListItem-secondaryAction-42"
      >
        <div
          class="MuiListItemText-root-44"
        >
          <span
            class="MuiTypography-root-50 MuiTypography-subheading-57 MuiListItemText-primary-47"
          >
            Two
          </span>
        </div>
      </div>
      <div
        class="MuiListItemSecondaryAction-root-86"
      />
    </li>
  </div>
  <div
    class="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
  >
    <li
      class="MuiListItem-container-33"
    >
      <div
        class="MuiListItem-root-32 MuiListItem-default-35 MuiListItem-gutters-40 MuiListItem-secondaryAction-42"
      >
        <div
          class="MuiListItemText-root-44"
        >
          <span
            class="MuiTypography-root-50 MuiTypography-subheading-57 MuiListItemText-primary-47"
          >
            Three
          </span>
        </div>
      </div>
      <div
        class="MuiListItemSecondaryAction-root-86"
      />
    </li>
  </div>
</ul>
`;
